pyogrio/_ogr.pyx

```diff
@@ -41,6 +41,23 @@ def get_gdal_version_string():
     return get_string(version)
 
 
+IF CTE_GDAL_VERSION >= (3, 4, 0):
+
+    cdef extern from "ogr_api.h":
+        bint OGRGetGEOSVersion(int *pnMajor, int *pnMinor, int *pnPatch)
+
+
+def get_gdal_geos_version():
+    cdef int major, minor, revision
+
+    IF CTE_GDAL_VERSION >= (3, 4, 0):
+        if not OGRGetGEOSVersion(&major, &minor, &revision):
+            return None
+        return (major, minor, revision)
+    ELSE:
+        return (0, 0, 0)
```

Comment:
I wonder if we should default this to `None` as well?  `None` for GDAL > 3.4 means GEOS is not installed whereas `None` here means we just can't tell.  But really the only thing we can use for debugging is when GEOS is available and GDAL > 3.4.

Reply:
I think we either know it or we don't know it.  Since we already can gather GDAL version, returning `(0, 0, 0)` simply shadows that, so I don't think it adds additional clarity.  We'll just have to remember to always be aware of GDAL version.